tests/Functional/PluginManagerCommandsTest.php

```diff
@@ -17,165 +17,160 @@ class PluginManagerCommandsTest extends TestCase
 
     /**
      * @test
-     * @covers \Pantheon\Terminus\Commands\Self\ListCommand
-     * @covers \Pantheon\Terminus\Commands\Self\ReloadCommand
-     * @covers \Pantheon\Terminus\Commands\Self\InstallCommand
-     * @covers \Pantheon\Terminus\Commands\Self\UpdateCommand
-     * @covers \Pantheon\Terminus\Commands\Self\UninstallCommand
-     * @covers \Pantheon\Terminus\Commands\Self\SearchCommand
+     * @covers \Pantheon\Terminus\Commands\Self\Plugin\ListCommand
+     * @covers \Pantheon\Terminus\Commands\Self\Plugin\ReloadCommand
+     * @covers \Pantheon\Terminus\Commands\Self\Plugin\InstallCommand
+     * @covers \Pantheon\Terminus\Commands\Self\Plugin\UpdateCommand
+     * @covers \Pantheon\Terminus\Commands\Self\Plugin\UninstallCommand
+     * @covers \Pantheon\Terminus\Commands\Self\Plugin\SearchCommand
      *
      * @group plugins
      * @group long
      */
-    public function testPluginsOperations()
+    public function testPluginsCommands()
     {
-
         $filesystem = new Filesystem();
         $pluginsDir = $this->getPluginsDir();
         $dependenciesBaseDir = $this->getDependenciesBaseDir();
         $filesystem->remove([$pluginsDir, $dependenciesBaseDir]);
 
-        // LIST COMMANDS TO CHECK THAT PLUGIN COMMANDS ARE NOT AVAILABLE
-        $command = 'hello';
-        $output = $this->terminus("list");
-        $this->assertStringNotContainsString($command, $output);
-
-        // LIST PLUGINS
-        $results = $this->terminusWithStderrRedirected("self:plugin:list");
+        // List commands to check that plugin commands are not available.
+        $helloCommand = 'hello';
+        $this->assertCommandDoesNotExist($helloCommand);
+        $this->assertNoPlugins();
+
+        // Search plugin.
+        $pluginName = 'terminus-plugin-example';
+        $pluginPackage = 'pantheon-systems/terminus-plugin-example';
+        $pluginList = $this->terminusJsonResponse(sprintf('self:plugin:search %s', $pluginName));
+        $this->assertIsArray($pluginList);
+        $this->assertNotEmpty($pluginList);
+        $plugin = reset($pluginList);
+        $this->assertIsArray($plugin);
+        $this->assertArrayHasKey('name', $plugin);
         $this->assertStringContainsString(
-            "You have no plugins installed",
-            $results,
-            "Terminus plugins should be empty at this point."
+            $pluginPackage,
```

Comment:
It's preferable to assert against constants rather than variables whenever possible.

Reply:
Even though you use this variable several times, I think it's better to switch back to constants in each place.

Reply:
@greg-1-anderson fixed